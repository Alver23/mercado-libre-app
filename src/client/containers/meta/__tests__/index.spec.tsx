// Dependencies
import { shallow, ShallowWrapper } from 'enzyme';

// Under test file
import Meta from '../index';

describe('<Meta />', () => {
  const props = {
    canonical: 'http://localhost/',
    title: 'fake title',
  };

  let component: ShallowWrapper;

  beforeEach(() => {
    component = shallow(<Meta {...props} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should save a snapshot of the component', () => {
    expect(component).toMatchSnapshot();
  });

  it('should render correctly when the prop is not sent image', () => {
    const newProps = {
      ...props,
      image: 'http://localhost:3000/img.png',
    };
    component.setProps(newProps);
    const ogImage = component.find('meta').filterWhere((element) => element.prop('content') === newProps.image);
    expect(ogImage).toBeTruthy();
  });
});
