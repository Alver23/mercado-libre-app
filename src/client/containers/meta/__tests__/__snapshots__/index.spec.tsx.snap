// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Meta /> should save a snapshot of the component 1`] = `
<Head>
  <meta
    charSet="UTF-8"
  />
  <meta
    content="width=device-width,initial-scale=1"
    name="viewport"
  />
  <link
    href="https://http2.mlstatic.com/frontend-assets/ui-navigation/5.14.3/mercadolibre/favicon.svg"
    key="favicon"
    rel="icon"
  />
  <title>
    fake title | undefined
  </title>
  <meta
    key="description"
    name="description"
  />
  <meta
    key="author"
    name="author"
  />
  <meta
    content="fake title | undefined"
    key="og:title"
    property="og:title"
  />
  <meta
    key="og:description"
    property="og:description"
  />
  <meta
    key="og:locale"
    property="og:locale"
  />
  <meta
    key="og:site_name"
    property="og:site_name"
  />
</Head>
`;
